### DISCUSSION TITLE
MiXCR and Cellranger output differences

### DISCUSSION THREAD

**Discussion by bshim181** [milaboratory/mixcr#1411] [Q&A] [answered]
I have a line of cellranger output which I notice differences between the MiXCR output and I am trying to understand why. 
This is the line of the cellranger output for the most abundant clonotype. 
<img width="915" alt="Screenshot 2023-09-19 at 2 56 41 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/e665dbf9-6d65-4706-8026-6538423b8516">
<img width="909" alt="Screenshot 2023-09-19 at 2 56 54 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/2263ae0c-b53a-477d-9975-e74ef065a34d">

I have searched this same CDR3 sequence in the MIXCR output and found about 300 different cell barcodes with UMIs evenly distributed among these cells. 
<img width="1382" alt="Screenshot 2023-09-19 at 2 57 47 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/e467d186-e7ce-4c83-b48a-5ce9224a6ea6">

I was wondering why these differences are observed. Is it because cellranger clonotype consensus report clusters clonotypes based on the CDR3 sequence similarity? (At the same time, I noticed duplicate CDR3 sequences in the entry, so I assumed this might not be the case). 

If this is the case, can a cell level report be formulated to output a CDR3 based clonotype reports? Also, is there functionality to extract representative TRA and TRB chains for each cell?

**Comment by mizraelson** [comment, marked as answer]
Hi,

To compare results, please use the `filtered_contig_annotations.csv` output file from CellRanger. In the file you provided (consensus_annotations.csv?), it appears the clones were not separated by cells. Multiple CDR3 entries might be a result of differences in the sequence outside CDR3, but its hard to tell without looking at the actual file.

MiXCR performes various correction and filtering steps to ensure the output TCR/BCR chains are verified. This means that in most cases, if two chains of the same type originate from a single cell ID, there would be a substantial number of reads (UMIs) confirming both. Two chains can indeed be biologically expressed, but contamination can sometimes occur, such as when two cells in a single droplet receive the same cell barcode.

Sincerely,
Mark
